rca/wagtailapi/filters.py

```diff
@@ -19,6 +22,43 @@ def filter_queryset(self, request, queryset, view):
         return queryset
 
 
+class SubjectsFilter(filters.BaseFilterBackend):
+    def filter_queryset(self, request, queryset, view):
+        if hasattr(queryset.model, "subjects"):
+            subject_ids = request.GET.getlist("subjects", [])
+            if subject_ids:
+                queryset = queryset.model.objects.filter(
+                    subjects__subject__in=subject_ids
+                )
+
+        return queryset
+
+
+class RelatedSchoolsFilter(filters.BaseFilterBackend):
+    def filter_queryset(self, request, queryset, view):
+        if hasattr(queryset.model, "related_schools_and_research_pages"):
+            school_ids = request.GET.getlist("related_schools_and_research_pages", [])
+            if school_ids:
+                school_pages = (
+                    SchoolsAndResearchPage.objects.all()
```

Comment:
`all` is redundant here. You can delete it. Do you not want to use `public()` too or check the permissions of the user?

Reply:
Yep - I'll remove `all()` I need to avoid using public though as the one schools and research page that will exist will not be available as public, but _will_ need to be used as a filter